pkg/controller/certificaterequests/ca/ca_test.go

```diff
@@ -391,3 +399,168 @@ func runTest(t *testing.T, test testT) {
+			})),
+			givenCR: gen.CertificateRequest("cr-1",
+				gen.SetCertificateRequestCSR(rsaCSR),
+				gen.SetCertificateRequestIssuer(cmmeta.ObjectReference{
+					Name:  "issuer-1",
+					Group: certmanager.GroupName,
+					Kind:  "Issuer",
+				}),
+			),
+			givenCASecret: gen.SecretFrom(gen.Secret("secret-1"), gen.SetSecretNamespace("default"), gen.SetSecretData(secretDataFor(t, rsaPair,
+				&x509.Certificate{
+					SerialNumber: big.NewInt(1234),
+					IsCA:         true,
+				},
+			))),
+			givenNamespace: "default",
+			assertSignedCert: func(t *testing.T, got *x509.Certificate) {
+				assert.Equal(t, []string{"http://ocsp-v3.example.org"}, got.OCSPServer)
+			},
+		},
+		"when the Issuer has crlDistributionPoints set, it should appear on the signed ca ": {
+			givenCAIssuer: gen.Issuer("issuer-1", gen.SetIssuerCA(cmapi.CAIssuer{
+				SecretName:            "secret-1",
+				CRLDistributionPoints: []string{"http://www.example.com/crl/test.crl"},
+			})),
+			givenCR: gen.CertificateRequest("cr-1",
+				gen.SetCertificateRequestIsCA(true),
+				gen.SetCertificateRequestCSR(rsaCSR),
+				gen.SetCertificateRequestIssuer(cmmeta.ObjectReference{
+					Name:  "issuer-1",
+					Group: certmanager.GroupName,
+					Kind:  "Issuer",
+				}),
+			),
+			givenCASecret: gen.SecretFrom(gen.Secret("secret-1"), gen.SetSecretNamespace("default"), gen.SetSecretData(secretDataFor(t, rsaPair,
+				&x509.Certificate{
+					SerialNumber: big.NewInt(1234),
+					IsCA:         true,
+				},
+			))),
+			givenNamespace: "default",
+			assertSignedCert: func(t *testing.T, gotCA *x509.Certificate) {
+				assert.Equal(t, []string{"http://www.example.com/crl/test.crl"}, gotCA.CRLDistributionPoints)
+			},
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			rec := &controllertest.FakeRecorder{}
+
+			c := &CA{
+				issuerOptions: controller.IssuerOptions{
+					ClusterResourceNamespace:        "",
+					ClusterIssuerAmbientCredentials: false,
+					IssuerAmbientCredentials:        false,
+				},
+				reporter: util.NewReporter(fixedClock, rec),
+				secretsLister: listers.FakeSecretListerFrom(listers.NewFakeSecretLister(),
+					listers.SetFakeSecretNamespaceListerGet(test.givenCASecret, nil),
+				),
+				templateGenerator: pki.GenerateTemplateFromCertificateRequest,
+			}
+
+			gotIssueResp, gotErr := c.Sign(context.Background(), test.givenCR, test.givenCAIssuer)
+			if test.wantErr != "" {
+				assert.EqualError(t, gotErr, test.wantErr)
+				return
+			}
+			require.NoError(t, gotErr)
+			require.NotNil(t, gotIssueResp)
+
+			gotCert, err := pki.DecodeX509CertificateBytes(gotIssueResp.Certificate)
+			require.NoError(t, err)
+
+			test.assertSignedCert(t, gotCert)
+		})
+	}
+}
+
+// Returns a map that is meant to be used for creating a certificate Secret
+// that contains the fields "tls.crt" and "tls.key".
+func secretDataFor(t *testing.T, caKey *rsa.PrivateKey, caCrt *x509.Certificate) (secretData map[string][]byte) {
```

Comment:
I imagine that in future there could be value in having this type of function available in some higher level test util package as we might reuse it elsewhere.